### ISSUE TITLE
Purchases: Update Payment Method for Multiple Purchases Together

### ISSUE THREAD

**Issue by donalirl** [Automattic/wp-calypso#40312]
Starting at URL: https://wordpress.com/me/purchases 

Similar to #38453, it would be helpful to be able to update the payment method used for multiple purchases at a time. 

This is a common source of confusion, and my report is prompted by a customer whose domain expired because of an invalid card and got hit with a $80 domain redemption fee. They had updated their payment method for their plan (a separate purchase) but didn't realize they needed to update the domain purchase separately.

It is unnecessarily tedious to follow this flow:

1. Click on their plan
2. Change the plan payment method
3. Go back to Purchases
4. Click on domain
5. Change the domain payment method

This can be especially tedious if a customer has many sites and purchases with us.

#### Context / Source
#2794008-zen  #painpoint #user-report

**Comment by kylemcph** [comment]
Noting that another user reached out in #21125175-hc and expressed the same frustration. They have eight separate subscriptions, and found it tedious and time consuming to update the expiring CC for each individual subscription. 

That also results in a very messy looking [Billing History](https://wordpress.com/me/purchases/billing) page, as the same card is now [added there eight times](https://neptunep2.files.wordpress.com/2020/09/2020-09-17-at-09.58.png)!


User provided us with the following feedback: 

```
“Update your help pages on payment options. Allow me to change my payment information ONCE for WordPress, not once for every product I have with you. Your happiness engineer did their best, but really, the payment system is not very good and needs thinking through from the CUSTOMER point of view, considering especially the case where the customer has a lot of products with you – which is what you want, right?”
```

**Comment by klimeryk** [comment]
@Automattic/shilling, has this been addressed with the recent work on redesigning these screens? (or maybe it's on your radar?)

**Comment by sirbrillig** [comment]
Thanks for the ping. There's several questions here. We do have a plan for improving the payment management for multiple subscriptions, and we've been working on it, first by updating the code for adding and changing payment methods, which is nearly done. After that we plan to add checkboxes to allow updating one or all subscriptions at once, followed eventually by a way to cherry-pick individual subscriptions to update at once. We also have a project to utilize Stripe card fingerprinting which should allow us to prevent storing the same credit card multiple times; otherwise this is actually quite difficult because we can't see the entire credit card number (for security). We can't know if the card `**** **** **** 4242` is the same as the card `**** **** **** 4242` since the first four digits could be different.

However, there's already one way to assign a new payment method to multiple subscriptions, although it's not ideal. Currently, if you click the "Add payment method" button and add a new card, that card will be applied to all of your subscriptions. That might help as a workaround until the rest of these projects can be completed.

**Comment by DavidRothstein** [comment]
>Currently, if you click the "Add payment method" button and add a new card, that card will be applied to all of your subscriptions.

Just to be clear, you do that by going to https://wordpress.com/me/purchases/payment-methods (i.e. the main "Payment methods" tab visible at the top of https://wordpress.com/me/purchases) and then click "Add payment method" from there.  Perhaps this needs to be documented better at https://wordpress.com/support/payment.

The main issues are that it isn't very discoverable, and also, if users **do** wind up on an individual subscription management page instead, there is no way to choose multiple subscriptions to add the card to from that page.

**Comment by sirbrillig** [comment]
> you do that by going to https://wordpress.com/me/purchases/payment-methods (i.e. the main "Payment methods" tab visible at the top of https://wordpress.com/me/purchases) and then click "Add payment method" from there.

Yes, and it also works via the "Add payment method" button at the site-level purchases page at  https://wordpress.com/purchases/payment-methods/example.com
